root/static/scripts/relationship-editor/components/RelationshipDialogContent.js

```diff
@@ -201,6 +205,57 @@ export function createInitialState(props: PropsT): RelationshipDialogStateT {
   };
 }
 
+/*
+ * Possibly updates newState.backward based on the source and target
+ * entities and the link type. The link direction is updated only in
+ * cases where the correct value can be inferred, e.g. "member of"
+ * relationships between person and group artist entities. See MBS-2604.
+ */
+function inferLinkDirection(newState: {...RelationshipDialogStateT}): void {
+  const source = newState.sourceEntity.source;
+  const target = newState.targetEntity.target;
+  const linkTypeId =
+    newState.linkType.autocomplete.selectedItem?.entity?.id ?? null;
+
+  if (
+    source.entityType === 'artist' &&
+    target.entityType === 'artist' &&
+    linkTypeId !== null &&
+    personGroupLinkTypeIds.has(linkTypeId)
+  ) {
+    const sourcePerson = source.typeID === ARTIST_TYPE_PERSON;
+    const sourceGroup =
+      source.typeID !== null && groupArtistTypeIds.has(source.typeID);
+
+    const targetPerson = target.typeID === ARTIST_TYPE_PERSON;
+    const targetGroup =
+      target.typeID !== null && groupArtistTypeIds.has(target.typeID);
+
+    if (sourcePerson && targetGroup) {
+      newState.backward = false;
+    } else if (sourceGroup && targetPerson) {
+      newState.backward = true;
+    }
+  }
+}
+
+// Artist types corresponding to groups.
+const groupArtistTypeIds = new Set([
```

Comment:
Would it be better to make this a generic helper function that may be more likely to get updated if new group subtypes are added in the future? If so, where should it be?

Reply:
This constant could be defined in `constants.js` instead.

Reply:
Sure, done.